docs/lib/getting-started/fragments/android/integrate.md

```diff
@@ -1,155 +1,459 @@
-);
-```
-
-d. Finally, you can receive notifications whenever a `Task` is created on the backend. To do so, start a new subscription:
-
-```java
-Amplify.API.subscribe(Task.class, SubscriptionType.ON_CREATE,
-    subscriptionId -> Log.i("AmplifyGetStarted", "Subscription established: " + subscriptionId),
-    taskCreated -> Log.i("AmplifyGetStarted", "Task created: " + taskCreated.getData().getTitle()),
-    apiFailure -> Log.e("AmplifyGetStarted", "Subscription failed.", apiFailure),
-    () -> Log.i("AmplifyGetStarted", "Subscription completed.")
-);
-```
-
-**Testing your API**
-The web-based AWS AppSync Console provides an easy way to run Queries, Mutation, or Subscription against your new API. The following command will directly open the Console for your API. When you run GraphQL operations on the server, you should be able to observe changes in your app.
-
-```bash
-amplify console api
-```
-
-When prompted, select **GraphQL**. This will open the AWS AppSync console for you to run Queries, Mutations, or Subscriptions at the server and see the changes in your client app.
-
-## Next Steps
-
-🎉 Congratulations! You have now built an app with a realtime backend.
-
-What next? Here are some things to add to your app:
-
-* [Analytics](~/lib/analytics/getting-started.md)
-* [API (GraphQL)](~/lib/graphqlapi/getting-started.md)
-* [API (REST)](~/lib/restapi/getting-started.md)
-* [Authentication](~/lib/auth/getting-started.md)
-* [DataStore](~/lib/datastore/getting-started.md)
-* [Storage](~/lib/storage/getting-started.md)
-
-**Existing AWS Resources**
-
-If you want to use your existing AWS resources with your app, you will need to **manually configure** your app by including relevant configurations into the `amplifyconfiguration.json` file.
-
-```json
-{
-  "UserAgent": "aws-amplify-cli/2.0",
-  "Version": "1.0",
-  "storage": {
-    "plugins": {
-      "awsS3StoragePlugin": {
-         "bucket": "my-s3-bucket",
-         "region": "us-west-2",
-         "defaultAccessLevel": "guest"
-      }
-    }
-  },
-  "analytics": {
-    "plugins": {
-      "awsPinpointAnalyticsPlugin": {
-        "pinpointAnalytics": {
-          "appId": "xxxx123xxxx23423bf24234",
-          "region": "us-east-1"
+Next you'll use the generated Todo model to create, update, query, and delete data. In this section you'll initialize DataStore, and then manipulate Todo items.
+
+## Configure Amplify and DataStore
+
+First, we'll add the DataStore plugin and configure Amplify. Typically, a good place to do this is in the [`onCreate()`](https://developer.android.com/reference/android/app/Application#onCreate()) method of [Android's `Application` class](https://developer.android.com/reference/android/app/Application). For the purposes of this tutorial, you can place these examples in the `onCreate()` method of the `MainActivity` class.
+
+1. Open `MainActivity` and add the following code to the bottom of the `onCreate()` method:
```

Comment:
It's not a best practice for a regular app to have this code in MainActivity but rather should go in the Application file - not sure if it's worth calling this out here since we had an issue related to this: https://github.com/aws-amplify/amplify-android/issues/211

Reply:
Hmm. I think in the `Project Setup` guide, we do it that way then. Maybe some weasel words about how we're doing this for demonstration purposes?